Description: Add support for x32
Author:
 Guillem Jover <guillem@debian.org>
 Daniel Schepler <dschepler@gmail.com>
Origin: vendor
Forwarded: no
Bug-Debian: 702183
Last-Update: 2020-04-11


---
 harness/main.c       |    2 +-
 src/syscall-x86_64.h |   10 ++++++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

--- a/src/syscall-x86_64.h
+++ b/src/syscall-x86_64.h
@@ -1,6 +1,16 @@
+#if __ILP32__
+#define __X32_SYSCALL_BIT    0x40000000
+#define __NR_io_setup (__X32_SYSCALL_BIT + 543)
+#define __NR_io_destroy (__X32_SYSCALL_BIT + 207)
+#define __NR_io_submit (__X32_SYSCALL_BIT + 544)
+#define __NR_io_cancel (__X32_SYSCALL_BIT + 210)
+#define __NR_io_getevents (__X32_SYSCALL_BIT + 208)
+#define __NR_io_pgetevents (__X32_SYSCALL_BIT + 333)
+#else
 #define __NR_io_setup		206
 #define __NR_io_destroy		207
 #define __NR_io_getevents	208
 #define __NR_io_submit		209
 #define __NR_io_cancel		210
 #define __NR_io_pgetevents	333
+#endif
--- a/harness/main.c
+++ b/harness/main.c
@@ -14,7 +14,7 @@
 #if __LP64__ == 0
 #if defined(__i386__) || defined(__powerpc__) || defined(__mips__)
 #define KERNEL_RW_POINTER	((void *)0xc0010000)
-#elif defined(__arm__) || defined(__m68k__) || defined(__s390__) || defined(__sh__)
+#elif defined(__arm__) || defined(__m68k__) || defined(__s390__) || defined(__sh__) || defined(__x86_64__)
 #define KERNEL_RW_POINTER	((void *)0x00010000)
 #elif defined(__hppa__)
 #define KERNEL_RW_POINTER	((void *)0x10100000)
